bats/tests/helpers/snapshots.bash

```diff
@@ -3,12 +3,13 @@ delete_all_snapshots() {
     assert_success
     # On Windows, executing native Windows executables consumes stdin.
     # https://github.com/microsoft/WSL/issues/10429
-    # Work around the issue by using `run` to populate `${lines[@]}` ahead of
-    # time, so that we don't need the buffer during the loop.
+    # Therefore, we have to collect all of the names before running any `rdctl`
+    # commands.  However, on macOS (bash 3.2), we seem to have issues with array
+    # variables; as a workaround, do an unquoted iteration.
     run jq_output .name
     assert_success
-    local name
-    for name in "${lines[@]}"; do
+    local name names=$output
+    for name in $names; do
```

Comment:
I don't think you need the additional `names` variable. Even if you would modify `output` inside the loop (which you don't), I don't see how the variable could be evaluated again inside the loop; it has to be done before the loop starts.

```suggestion
    local name
    for name in $output; do
```

However, this changes assumes that snapshot names do not include whitespace, which is not guaranteed. See `bats/tests/test-snapshot-list.bats` for examples.